src/main/java/frc/trigon/robot/commands/factories/ShootingCommands.java

```diff
@@ -71,7 +78,11 @@ private static Command getPrepareCloseSpeakerShotCommand() {
     }
 
     private static Command getFeedNoteForCloseSpeakerShotCommand() {
-        return GeneralCommands.runWhen(IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.FEED_SHOOTING), () -> RobotContainer.SHOOTER.atTargetVelocity() && RobotContainer.PITCHER.atTargetPitch());
+        return GeneralCommands.runWhen(IntakeCommands.getSetTargetStateCommand(IntakeConstants.IntakeState.FEED_SHOOTING).alongWith(getVisualizeNoteShootingCommand()), () -> RobotContainer.SHOOTER.atTargetVelocity() && RobotContainer.PITCHER.atTargetPitch());
+    }
+
+    public static Command getVisualizeNoteShootingCommand() {
+        return new InstantCommand(() -> GeneralCommands.runWhen(new VisualizeNoteShootingCommand(), () -> SimulationObjectDetectionCameraIO.HAS_OBJECTS).schedule()).onlyIf(() -> RobotHardwareStats.isSimulation());
```

Comment:
We schedule this if the robot isn't real (this can also happen in replay), so we can replay the robot to get this as well. 
Once Nahum finished the whole mechanism only logged if chosen so in the dashboard, the logic should be this way: 
If we are not in replay, only schedule if extensive logging is enabled. If we are in replay, always make this happen.